### ISSUE TITLE
declare -x -f my_function is mistaken for an unassigned variable

### ISSUE THREAD

**Issue by per-bohlin** [koalaman/shellcheck#1209]
#### For bugs
- Rule Id: SC2154
- version: 0.4.6
- [x ] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/bash

my_expensive_function() {
    echo "my super expensive long running function"
}

declare -x -f my_expensive_function

nohup /bin/bash -c "my_expensive_function" &>> out.txt < /dev/null &
```

#### Here's what shellcheck currently says:
```
declare -x -f my_expensive_function
              ^-- SC2154: my_expensive_function is referenced but not assigned.
```

#### Here's what I wanted or expected to see:
I would assume exporting an existing function to be valid in bash (running bash 4.3.48).

**Comment by matthewpersico** [comment]
s/declare -x -f my_expensive_function/export -f my_expensive_function/

declare by itself shows all variables and functions in the env
declare -f means show only the functions in the env
declare -x marks the variable for export
Used together, not sure what -x and -f so when used together
declare does not export

I think...

**Comment by per-bohlin** [comment]
At least in bash 4.3.38 on ubuntu 16.04 it does.

declare -f 
lists all functions, including definition

declare -f my_function
lists all functions, including definition

declare -x -f my_function
doesn't print anything but exports the symbol "my_function" and hence the function body.

To try it out.

```
$ foo() { echo "foo"; }
$ bash -c "foo"
bash: foo: command not found
$ declare -f foo
foo () 
{ 
    echo foo
}
$ bash -c "foo"
bash: foo: command not found
$ declare -x -f foo
$ bash -c "foo"
foo
```

**Comment by per-bohlin** [comment]
But you are right, export -f has the same effect and is more explicit with its intent. I will use that from now on.

**Comment by matthewpersico** [comment]
Hmmm. I tested on my WSL instance. Will try at work.

**Comment by koalaman** [comment]
Plain ol' bug. It's now fixed. Thanks!
